metaflow/client/core.py

```diff
@@ -570,6 +622,48 @@ def traverse(cls, ids_r, lst):
             self._path_components = traverse(_CLASSES[self._NAME], ids, [])
         return self._path_components
 
+    def add_tag(self, tags):
+        """
+        Add one or more tags to the current object
+
+        Parameters
+        ----------
+        tags : string or List[string]
+            Tags to add
+        """
+        self.replace_tag([], tags)
+
+    def remove_tag(self, tags):
+        """
+        Removes one or more tags to the current object
+
+        Parameters
+        ----------
+        tags : string or List[string]
+            Tags to remove
+        """
+        self.replace_tag(tags, [])
+
+    def replace_tag(self, remove_tags, add_tags):
+        """
+        Removes and adds tags; the removal is done first
+
+        Parameters
+        ----------
+        remove_tags : string or List[string]
+            Tags to remove
+        add_tags : string or List[string]
+            Tags to add
+        """
+        if is_stringish(remove_tags):
+            remove_tags = [remove_tags]
+        if is_stringish(add_tags):
+            add_tags = [add_tags]
+        operations = [('remove', tag) for tag in remove_tags]
```

Comment:
you should check that each `tag` `is_stringish` - currently `add_tag([1])` happily adds an integer as a tag